File: leetCode/fukun/lc_083.js
function ListNode(val) {
    this.val = val;
    this.next = null;
}

const deleteDuplicates = function (head) {
    if(!head)return head;
    let curr = head;
    let hash = {};
    hash[head.val] = true;
    while (curr && curr.next) {
        if (hash[curr.next.val]) {
            curr.next = curr.next.next;
        } else {
            hash[curr.next.val] = true;
            curr = curr.next;
        }
    }
    return head;
};

let ln0 = new ListNode(0);
let ln1 = new ListNode(1);
let ln2 = new ListNode(1);
ln0.next = ln1;
ln1.next = ln2;
console.log(deleteDuplicates(ln0));

File: leetCode/XSS/lc_083.java
public class lc_083 {
    public static void main(String[] args) {
//删除排序链表中的重复元素，*
    }

      public class ListNode {
          int val;
          ListNode next;
          ListNode(int x) { val = x; }
      }
    static class Solution {
        public ListNode deleteDuplicates(ListNode head) {
            if(head==null||head.next==null)return head;
            ListNode p=head.next;
            ListNode pp=head;
            int preval=head.val;
            while(p!=null)
            {
                if(p.val==preval)
                {
                    pp.next=p.next;
                }
                else {
                    pp=p;
                    preval=p.val;
                }
                p=p.next;
            }
            return head;
        }
    }
}
